teps/0108-mapping-workspaces.md

```diff
@@ -0,0 +1,119 @@
+---
+status: implementable
+title: Mapping Workspaces
+creation-date: '2022-05-03'
+last-updated: '2022-05-03'
+authors:
+- '@jerop'
+- '@bobcatfish'
+see-also:
+- TEP-0107
+---
+
+# TEP-0108: Mapping Workspaces
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This proposal builds on this prior work to reduce verbosity in mapping `Workspaces` and improve usability of
+*Tekton Pipelines*.
+
+## Motivation
+
+The verbosity of writing specifications in *Tekton Pipelines* is a common pain point that causes difficulties in 
+getting-started scenarios. `Tasks` declare `Workspaces` they need, while `Pipelines` declare `Workspaces` that are
+shared among its `PipelineTasks`. The mapping of `Workspaces` from `Pipelines` to `PipelineTasks` is verbose, as
+shown below:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: pipeline
+spec:
+  workspaces:
+    - name: source
+  tasks:
+    - name: gen-code
+      taskRef:
+        name: gen-code # gen-code expects a Workspace named "source"
+      workspaces:
+        - name: source
+          workspace: source
+    - name: commit
+      taskRef:
+        name: commit # commit expects a Workspace named "source"
+      workspaces:
+        - name: source
+          workspace: source
+      runAfter:
+        - gen-code
+```
+
+## Proposal
+
+We propose auto-mapping `Workspaces` from `Pipelines` to `PipelineTasks` when the names of the `Workspaces` declared in
+the `Pipeline` and `PipelineTask` are the same, as shown below:
```

Comment:
I am assuming we will implement validation routine to confirm the workspaces required by the `tasks` are specified in the `pipeline` and fail if such workspaces (without mapping) are missing from the `pipeline`.

Reply:
yes, the workspaces validation that checks workspaces that are needed exist will be expanded to cover this

Reply:
updated the TEP to include this detail

Reply:
updated the TEP to include this detail